### PR TITLE
mgmt: Move mcumgr into its own folder

### PR DESCRIPTION
**Pull request by carlescufi**
In order to be able to add more entries under 'subsys/mgmt', move the
current contents of it, which relate exclusively to MCUMgr, to its own
folder.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

**Comment by carlescufi** [comment]
@nandojve once this is merged, could you please send a PR to move all the code in `lib/updatehub` to `subsys/mgmt/updatehub`?

**Comment by nashif** [comment]
we probably want to move the headers as well, right now include/mgmt is all about mcumgr, i.e. 'include/mgmt/buf.h'

**Comment by carlescufi** [comment]
> we probably want to move the headers as well, right now include/mgmt is all about mcumgr, i.e. 'include/mgmt/buf.h'

yes, good point. I will do that as well.

**Comment by nandojve** [comment]
> @nandojve once this is merged, could you please send a PR to move all the code in `lib/updatehub` to `subsys/mgmt/updatehub`?

Yes, I can work on it.
CC: @otavio

File: subsys/mgmt/CMakeLists.txt
Comment (de-nordic) [inline subsys/mgmt/CMakeLists.txt:3]: This could be probably replaced with `add_subdirectory_if_kconfig`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Need to update `MAINTAINERS.yml` too

**Comment by carlescufi** [comment]
> Need to update `MAINTAINERS.yml` too

Indeed. @nashif do you think we should tweak the compliance script so it checks for required changes in `MAINTAINERS.yml`?

**Comment by carlescufi** [comment]
> Need to update `MAINTAINERS.yml` too

Done now, @MaureenHelm

File: subsys/mgmt/CMakeLists.txt
Comment (de-nordic) [reply subsys/mgmt/CMakeLists.txt:3]: Good point.

Comment (carlescufi) [reply subsys/mgmt/CMakeLists.txt:3]: @tejlmand should we deprecate the `if_kconfig()` in favor of  `_ifdef` ?

Comment (de-nordic) [reply subsys/mgmt/CMakeLists.txt:3]: @carlescufi Currently there are 42 instances of `add_subdirectory_if_kconfig` in CMakeLists.txt of Zephyr. I think that we could RFC the change. I have looked into the macro and thought about it a little and I think that it is not needed as it is just specialized `_ifdef`. When change to the config identifier happens or the file name gets changed you may end up replacing the macro, in CMakeLists.txt` with the `_ifdef` variant anyway.

Comment (carlescufi) [reply subsys/mgmt/CMakeLists.txt:3]: Agreed, I think we should remove it altogether and replace all instances with `_ifdef`. We make no promises of backwards compatibility in CMake macros or functions, so there's no need for deprecation here.

**Review by de-nordic — APPROVED** [review]
I am fine with the change.
Thanks!

**Review by nashif — APPROVED** [review]

File: subsys/mgmt/CMakeLists.txt
Comment (carlescufi) [reply subsys/mgmt/CMakeLists.txt:3]: #27290

**Review by MaureenHelm — APPROVED** [review]

**Comment by pabigot** [comment]
We can argue whether moving include files so existing code doesn't build is an API change, but I certainly think there should be an entry in the release notes explaining why it was done.

**Pull request merged**
